python/mead/tf/exporters.py

```diff
@@ -186,18 +207,16 @@ def _create_preprocessed_input(self, tf_example,
 
         return preprocessed, lengths
 
-    def _get_max_lens(self, base_name):
-        mxlen = self.task.config_params['preproc']['mxlen']
-        mxwlen = self.task.config_params['preproc'].get('mxwlen')
-        state = baseline.utils.read_json("{}.state".format(base_name))
-        if 'mxlen' in state:
-            mxlen = state['mxlen']
-        # What should be called mxwlen is called maxw in the state object of this is for backwards compatibility.
-        if 'maxw' in state:
-            mxwlen = state['maxw']
-        if 'mxwlen' in state:
-            mxwlen = state['mxwlen']
-        return mxlen, mxwlen
+    def _get_max_lens(self, feature_descs):
+        """
+        _get_max_lens() restructures the feature_descs object
+        to order them for mxlen and mxwlen, which is word and char.
+        """
+        if 'word' not in feature_descs:
```

Comment:
Its actually possible to not have a feature named 'word' or 'char' at all.  There is a concept of a primary feature when mead runs now, and the vectorizers also track both the mxlen/mxwlen as well as any user-defined length constraints, so I think this can be generalized.